Userland/wc.cpp

```diff
@@ -2,143 +2,131 @@
 #include <AK/Vector.h>
 #include <LibCore/CArgsParser.h>
 
-#include <ctype.h>
 #include <stdio.h>
+#include <sys/stat.h>
 
-static bool output_chars = false;
-static bool output_words = false;
-static bool output_lines = false;
+struct WhatToPrint {
+    bool line_flag=false;
+    bool byte_flag=false;
+    bool char_flag=false;
+    bool word_flag=false;
+};
 
-struct Count {
-    String file;
-    unsigned long chars = 0;
-    unsigned long words = 0;
-    unsigned long lines = 0;
+struct Word {
+    bool line=false;
+    bool tab=false;
+    bool space=false;
 };
 
-int wc(const String& filename, Vector<Count>& counts);
+struct Count {
+    unsigned int lines=0;
+    unsigned int characters=0;
+    unsigned int words=0;
+    size_t bytes=0;
+};
 
-void report(const Count& count)
+Vector<String> get_files(int argc, char** argv)
 {
-    if (output_lines) {
-        printf("%lu ", count.lines);
-    }
-    if (output_words) {
-        printf("%lu ", count.words);
-    }
-    if (output_chars) {
-        printf("%lu ", count.chars);
-    }
-    printf("%s\n", count.file.characters());
+    Vector<String> files;
+    for (int i=1; i<argc; i++)
+    	if (argv[i][0]!='-')
+            files.append(argv[i]);
+    return files;
 }
 
-void report(const Vector<Count>& counts)
+void wc_out(WhatToPrint& flags, Count& count, String name)
 {
-    Count total { "total" };
-    for (const auto& c : counts) {
-        report(c);
-        total.lines += c.lines;
-        total.words += c.words;
-        total.chars += c.chars;
-    }
-    if (counts.size() > 1) {
-        report(total);
-    }
-    fflush(stdout);
-}
+    if (flags.line_flag)
+        printf("%7i ", count.lines);
+    if (flags.word_flag)
+        printf("%7i ", count.words);
+    if (flags.byte_flag)
+        printf("%7lu ", count.bytes);
+    if (flags.char_flag)
+        printf("%7i ", count.characters);
 
-int count_words(const char* s)
-{
-    int n = 0;
-    bool in_word = false;
-    for (; *s; ++s) {
-        if (!isspace(*s)) {
-            if (!in_word) {
-                in_word = true;
-                ++n;
-            }
-        } else if (in_word) {
-            in_word = false;
-        }
-    }
-    return n;
+    printf("%14s\n", name.characters());
 }
 
-int main(int argc, char** argv)
+int main(int argc, char **argv)
 {
-    if (argc < 2) {
-        printf("usage: wc [-c|-m] [-lw] [file...]\n");
-        return 0;
-    }
+    struct stat st;
```

Comment:
This isn't C. Please don't declare variables unnecessarily early. Declare them at first use instead :)